### PR TITLE
haskellPackages.streamly-archive: unbreak

### PR DESCRIPTION
**Pull request by shlok**
###### Motivation for this change

The correct libarchive is now being passed into the package.

###### Things done

- [x] Tested compilation of all pkgs that depend on this change using `nix-build --no-out-link -A haskellPackages.streamly-archive --arg config '{ allowBroken = true; }'`
- [x] Ran `maintainers/scripts/haskell/regenerate-hackage-packages.sh` and included the result in the PR.

### CODE REVIEW

**Comment by maralorn** [comment]
@shiok If you want you can run `maintainers/scripts/haskell/regenerate-haskell-packages.sh` to immediately see and test the consequence of your change.

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -1867,4 +1867,7 @@ self: super: {
   # https://github.com/travitch/haggle/issues/4
   haggle = doJailbreak super.haggle;
 
+  # Pass the correct libarchive into the package.
+  streamly-archive = super.streamly-archive.override { archive = pkgs.libarchive; };
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-common.nix:1871]: This override should go into `configuration-nix.nix`.

**Comment by maralorn** [comment]
This builds fine for me. If you sqash this I can merge it. I would love if you could run the above mentioned script from the root of your nixpkgs repo as part of your commit. But I can run it afterwards if you want.

**Comment by shlok** [comment]
@maralorn Thanks; I have squashed the commits now. Also included is the result of `regenerate-hackage-packages.sh` (although FYI, I had to do `export LC_ALL=C.UTF-8` to avoid a certain `commitBuffer: invalid argument (invalid character)` error on my Debian 10).

**Comment by maralorn** [comment]
@shlok Thank you for testing the script!

Huh, that seems worth investigating. Do you think it would be smart to set that variable in the beginning of the script to prevent this from happening for other users?

**Comment by shlok** [comment]
> @shlok Thank you for testing the script!

My pleasure :)

> Do you think it would be smart to set that variable in the beginning of the script to prevent this from happening for other users?

I'm not sure about that. For me it was a mere workaround based on an internet search that led me to https://github.com/commercialhaskell/stack/issues/793

**Comment by maralorn** [comment]
Shows to double check, if you can tell: Does the fail happen while creating the nix-shell the script starts or this an error that hackage2nix throws?

**Comment by shlok** [comment]
It's thrown by `hackage2nix`:

    hackage2nix: /home/user/nixpkgs/pkgs/development/haskell-modules/hackage-packages.nix: commitBuffer: invalid argument (invalid character)

(I added `export LC_ALL=C.UTF-8` just before the `hackage2nix` line to "fix" the issue.)

By the way, I just noticed that `LC_ALL` is not set at all before my `export` line. It however is set on my local machine, so I guess this is a nix-shell thing.

**Comment by cdepillabout** [comment]
I imagine that this LC_ALL=C.UTF-8 issue is the same as https://github.com/NixOS/nixpkgs/issues/64603#issuecomment-551419489.

We could probably set `LC_ALL=C.UTF-8` in `regenerate-hackage-packages.sh` for users that don't have this already set on their system.

**Comment by maralorn** [comment]
I tried adressing this in [`66de11d` (#122117)](https://github.com/NixOS/nixpkgs/pull/122117/commits/66de11da71cf3de9a748eb30cad667e6147b4c41)

**Pull request merged**